core/src/main/java/io/zeebe/containers/ZeebeVolume.java

```diff
@@ -160,9 +164,14 @@
    */
   public static ZeebeVolume newVolume(final UnaryOperator<CreateVolumeCmd> configurator) {
     final DockerClient client = DockerClientFactory.instance().client();
-    try (final CreateVolumeCmd command = client.createVolumeCmd()) {
+    final Map<String, String> labels = new HashMap<>();
+    labels.putAll(DockerClientFactory.DEFAULT_LABELS);
+    //noinspection deprecation
+    labels.putAll(ResourceReaper.instance().getLabels());
```

Comment:
I guess this means it's at risk of being removed, or at least hidden? :thinking:

Reply:
Instead of relying on an internal API, what about adding our own marker label? So we keep adding the default labels, but then add our own label (e.g. `io.camunda.zeebe.sessionId`), and register it via 

```java
public final class ZeebeVolume {
  private static final String SESSION_LABEL = "io.zeebe.containers.sessionId";
  private static final Map<String, String> MARKER_LABELS = Collections.singletonMap(SESSION_LABEL, DockerClientFactory.SESSION_ID);

  static {
    ResourceReaper.instance().registerLabelsFilterForCleanup(MARKER_LABELS);
  }

  public static ZeebeVolume newVolume(final UnaryOperator<CreateVolumeCmd> configurator) {
    final DockerClient client = DockerClientFactory.instance().client();
    try (final CreateVolumeCmd command = client.createVolumeCmd()) {
      final Map<String, String> labels = new HashMap<String, String>(DockerClientFactory.DEFAULT_LABELS);
      labels.putAll(MARKER_LABELS);

      final CreateVolumeResponse response = configurator.apply(command.withLabels(labels).exec();
      return new ZeebeVolume(response.getName(), client);
    }
  }
```

Reply:
This also means we won't need reflection in the test to check if everything is working as expected.

Reply:
Ah right, that's how this is supposed to work :facepalm: 

Although I don't see how this removes the need for reflection in the tests. We'd still want to test that the container is actually removed, right?

Reply:
Wouldn't `ResourceReaper.instance().performCleanup()` work? It's deprecated, but here it's only for tests :shrug:

Reply:
I'm a bit wary of whether my suggestion works with Ryuk though. It should, it seems like Ryuk receives new configuration over time, but I haven't tested it.

Reply:
Doesn't work, I think because `ResourceReaper.instance()` is a `RyukResourceReaper` by default which doesn't override `performCleanup` and the default implementation only removes containers (and all of its attached volumes, regardless of their labels) that were registered by `registerContainerForCleanup` before.

Reply:
Ah yeah, because Ryuk itself will clean volumes, so it doesn't have to do that :smile: `instance` seems to instantiate a `JVMHookResourceReaper` only if the env var is set, which you can't override per test (except with some voodoo) :(

Reply:
Yes, that's exactly what `getCleanupRunnable` does, right?

Reply:
I thought maybe that part is not necessary?

```
        final Method performCleanup = reaperClass.getDeclaredMethod("performCleanup");
        performCleanup.setAccessible(true);
        performCleanup.invoke(reaper);
```

:shrug: It's really minor anyway

Reply:
Ah yeah that works :+1:

Reply:
Registering the new labels with Ryuk does work btw:
```
[testcontainers-ryuk] DEBUG org.testcontainers.utility.ResourceReaper - Sending 'label=io.zeebe.containers.sessionId%3D85662216-fce6-4aab-a7c1-fe9afd3e3419' to Ryuk
```

Reply:
Cool, we can make use of that when they remove the deprecated API :smile: